From 11174408e833cb53dd532666d61f028a94fc48d5 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 8 Feb 2013 14:16:10 +0100
Subject: [PATCH] systemd/dracut-initqueue.sh: replace $UDEV_QUEUE_EMPTY

for systemd-udevd the version of udevd is high enough
---
 modules.d/98systemd/dracut-initqueue.sh | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/modules.d/98systemd/dracut-initqueue.sh b/modules.d/98systemd/dracut-initqueue.sh
index c82b6eb..76160cb 100755
--- a/modules.d/98systemd/dracut-initqueue.sh
+++ b/modules.d/98systemd/dracut-initqueue.sh
@@ -39,7 +39,7 @@ while :; do
         check_finished && break 2
     done
 
-    $UDEV_QUEUE_EMPTY >/dev/null 2>&1 || continue
+    udevadm settle --timeout=0 >/dev/null 2>&1 || continue
 
     for job in $hookdir/initqueue/settled/*.sh; do
         [ -e "$job" ] || break
@@ -47,12 +47,11 @@ while :; do
         check_finished && break 2
     done
 
-    $UDEV_QUEUE_EMPTY >/dev/null 2>&1 || continue
+    udevadm settle --timeout=0 >/dev/null 2>&1 || continue
 
     # no more udev jobs and queues empty.
     sleep 0.5
 
-
     if [ $main_loop -gt $((2*$RDRETRY/3)) ]; then
         for job in $hookdir/initqueue/timeout/*.sh; do
             [ -e "$job" ] || break
